src/lib/controllers/ui-controller.js

```diff
@@ -66,9 +67,46 @@ export default class UIController {
     this.isActivated = false
     this.isDeactivated = false
 
+    /**
+     * If an event controller be used with an instance of a UI Controller,
+     * this prop will hold an event controller instance. It is usually initialized within a `build` method.
+     * @type {UIEventController}
+     */
+    this.evc = null
+
     this.inflectionsViewSet = null // Holds inflection tables ViewSet
   }
 
+  /**
+   * Creates an instance of a UI controller with default options. Provide your own implementation of this method
+   * if you want to initialize a UI controller differently.
+   */
+  static build (state, options) {
+    let uiController = new UIController(state, options)
+
+    // Creates on configures an event listener
+    let eventController = new UIEventController()
+    switch (uiController.options.textQueryTrigger) {
+      case 'dblClick':
+        eventController.registerListener('GetSelectedText', uiController.options.textQuerySelector, uiController.getSelectedText.bind(uiController), MouseDblClick)
+        break
+      case 'longTap':
+        eventController.registerListener('GetSelectedText', uiController.options.textQuerySelector, uiController.getSelectedText.bind(uiController), LongTap)
+        break
+      default:
+        eventController.registerListener(
+          'GetSelectedText', uiController.options.textQuerySelector, uiController.getSelectedText.bind(uiController), GenericEvt, uiController.options.textQueryTrigger
+        )
+    }
+
+    eventController.registerListener('HandleEscapeKey', document, uiController.handleEscapeKey.bind(uiController), GenericEvt, 'keydown')
```

Comment:
FYI, this is a pre-existing problem but I think hard-coded use of the `document` object is problematic here as is use of `body` in the AlpheiosPageLoad listener. These should really be coming from the uiController constructor arguments.

Reply:
Could you please describe what situation would require to handle it differently? Is it something related to the way UI controller be used?

Reply:
I was thinking of a case where the main document object is not what Alpheios is bound to. It's possible we don't have a need for this, but with the embedded library I did not want to force any assumptions of how and where it might be used.